### PR TITLE
Eliminate guards through max_pool ops.

### PR DESCRIPTION
**Pull request by resistor**

### CODE REVIEW

File: torch/csrc/jit/passes/guard_elimination.cpp [hunk 1]
```diff
@@ -284,6 +284,18 @@ struct GuardElimination {
...
+    case aten::max_pool3d:
+      return !n->input(0)->type()->expect<TensorType>()->isSummarized() &&
+          // check that the kernel size is constant
+          n->input(1)->node()->kind() == prim::Constant &&
+          // check that the stride is constant
+          n->input(2)->node()->kind() == prim::Constant &&
+          // check that the padding is constant
+          n->input(3)->node()->kind() == prim::Constant &&
+          // check that the dilation is constant
+          n->input(4)->node()->kind() == prim::Constant;
```
Comment (Krovatkin) [inline torch/csrc/jit/passes/guard_elimination.cpp]: `ceil_mode` could also potentially affect output shapes, shouldn't we have another check on `input(5)`?

**Review by Krovatkin — CHANGES_REQUESTED** [review]
could we also add a simple test?

File: torch/csrc/jit/passes/guard_elimination.cpp (hunk 1 above)
Comment (resistor) [reply torch/csrc/jit/passes/guard_elimination.cpp]: I don't see how `ceil_mode` could affect output shapes?

Comment (resistor) [reply torch/csrc/jit/passes/guard_elimination.cpp]: Fixed

**Comment by resistor** [comment]
> could we also add a simple test?

Added

**Review by Krovatkin — APPROVED** [review]

**Pull request closed without merging**